src/vl-map-wmts-layer.js

```diff
@@ -0,0 +1,69 @@
+import {VlMapLayer} from '/src/vl-map-layer.js';
+import {OlWMTSSource, OlWMTSTileGrid, OlTileLayer, OlExtent} from '/node_modules/vl-mapactions/dist/vl-mapactions.js';
+/**
+ * VlMapWmtsLayer
+ *
+ * @class
+ * @classdesc Een WMTS (overlay) layer
+ *
+ * @extends VlMapBaseLayer
+ *
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/releases/latest|Release notes}
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/issues|Issues}
+ * @see {@link https://webcomponenten.omgeving.vlaanderen.be/demo/vl-map-wmts-layer.html|Demo}
+ */
+export class VlMapWmtsLayer extends VlMapLayer {
+  constructor() {
+    super();
+    this._layer = this._createWmtsLayer();
+  }
+
+  async connectedCallback() {
+    await super.connectedCallback();
+  }
+
+
+  get _projection() {
+    if (this.parentNode) {
+      return this.parentNode._projection;
+    }
+  }
+
+  get url() {
+    return this.getAttribute('data-vl-url');
+  }
+
+  _createWmtsLayer() {
+    const size = OlExtent.getWidth(this._projection.getExtent()) / 256;
+    const resolutions = new Array(16);
+    const matrixIds = new Array(16);
+    for (let z = 0; z < 16; ++z) {
+      resolutions[z] = size / Math.pow(2, z);
+      matrixIds[z] = z;
+    }
+
+    const source = new OlWMTSSource({
+      url: this.url,
+      layer: this.getAttribute('data-vl-layer'),
```

Comment:
Ineens een getter maken voor de layer?

Reply:
op vl-map-layer niveau dan

Reply:
Gewoon die `this.getAttribute('data-vl-layer')` extracten naar een getter bedoelde ik.

Reply:
Zal dit even extracten